vnext/ReactWindowsCore/utilities.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+#include <type_traits>
+#include <utility>
+
+namespace facebook {
+namespace react {
+namespace utilities {
+
+// A simple wrapper for reinterpret_casting from TOriginalTypePtr to
+// TResultTypePtr that does a static_assert before performing the cast. The
+// usage syntax for this function is the same as the syntax for doing a regular
+// reinterpret_cast. For example:
+//
+//   uint32_t* i;
+//   auto p = checkedReinterpretCast<char32_t*>(i);
+//
+template <typename TResultTypePtr, typename TOriginalTypePtr>
+inline TResultTypePtr checkedReinterpretCast(TOriginalTypePtr p) noexcept
+{
+  using TResultType = typename std::remove_pointer<TResultTypePtr>::type;
+  using TOriginalType = typename std::remove_pointer<TOriginalTypePtr>::type;
+
+  static_assert(
+    std::is_pointer<TResultTypePtr>::value &&
+    std::is_pointer<TOriginalTypePtr>::value &&
+    std::is_integral<TResultType>::value &&
+    std::is_integral<TOriginalType>::value &&
+    sizeof(TResultType) == sizeof(TOriginalType),
+    "checkedReinterpretCast can only be used to cast from T1* to T2*, where "
+    "T1 and T2 are integral types of the same size.");
+
+  return reinterpret_cast<TResultTypePtr>(p);
+}
+
+// A compile time function that deduces the size of an array.
+template<typename T, std::size_t N>
+constexpr std::size_t arraySize(T(&)[N]) noexcept
```

Comment:
>arraySize [](start = 22, length = 9)

This function is not yet used anywhere in the code base. However, a similar function exists in the ChakraJsiRuntime implementations. This function will also be used by some unicode conversion tests that I plan on checking in later as a separate PR.